openbb_platform/providers/intrinio/openbb_intrinio/models/company_news.py

```diff
@@ -61,12 +62,21 @@ def extract_data(
     ) -> List[Dict]:
         """Return the raw data from the Intrinio endpoint."""
         api_key = credentials.get("intrinio_api_key") if credentials else ""
+        results: List[Dict] = []
 
-        base_url = "https://api-v2.intrinio.com/companies"
-        query_str = get_querystring(query.model_dump(by_alias=True), ["symbols"])
-        url = f"{base_url}/{query.symbols}/news?{query_str}&api_key={api_key}"
+        def get_data(symbol):
+            base_url = "https://api-v2.intrinio.com/companies"
+            query_str = get_querystring(query.model_dump(by_alias=True), ["symbols"])
+            url = f"{base_url}/{symbol}/news?{query_str}&api_key={api_key}"
+            data = get_data_many(url, "news", **kwargs)
+            data = [{**d, "symbol": symbol} for d in data]
+            results.append(data)
 
-        return get_data_many(url, "news", **kwargs)
+        with ThreadPoolExecutor() as executor:
+            executor.map(get_data, [s.strip() for s in query.symbols.split(",")])
+
+        results = [item for sublist in results for item in sublist]
```

Comment:
What is the use of this line of code?

Reply:
It makes sure that we don't end up with a List[List[Dict]]